File: src/test/java/com/github/qky666/selenidepom/test/java/mtp/pom/kotlinpom/CookiesBannerWidget.java
package com.github.qky666.selenidepom.test.java.mtp.pom.kotlinpom;

import com.codeborne.selenide.Condition;
import com.codeborne.selenide.SelenideElement;
import com.github.qky666.selenidepom.annotation.Required;
import com.github.qky666.selenidepom.pom.Widget;
import lombok.Getter;

import java.time.Duration;

import static com.codeborne.selenide.Condition.disappear;
import static com.github.qky666.selenidepom.pom.LoadableKt.shouldLoadRequired;

public class CookiesBannerWidget extends Widget {

    // Fields
    @Getter(lazy = true) @Required private final SelenideElement cookiesText = this.$("div.cli-bar-message");
    @Getter(lazy = true) @Required private final SelenideElement accept = this.$("a#cookie_action_close_header");

    // Constructor
    public CookiesBannerWidget(SelenideElement self) {super(self);}

    // Methods
    @Override
    public void customShouldLoadRequired(Duration timeout, String pomVersion) throws Throwable {
        super.customShouldLoadRequired(timeout, pomVersion);
        getCookiesText().shouldHave(Condition.text("Utilizamos cookies propias y de terceros para fines analíticos y para mostrarte publicidad personalizada en base a un perfil elaborado a partir de tus hábitos de navegación (por ejemplo, páginas visitadas)"), timeout);
    }

    public void acceptCookies() {
        shouldLoadRequired(this).getAccept().click();
        this.should(disappear);
    }
}

File: cookiecutter/{{cookiecutter.repo_name}}/src/test/kotlin/{{ cookiecutter.__group_folder }}/pom/common/CookiesBannerWidget.kt
@file:Suppress("unused")

package {{ cookiecutter.group }}.pom.common

import com.codeborne.selenide.Condition.disappear
import com.codeborne.selenide.Condition.text
import com.codeborne.selenide.SelenideElement
import com.github.qky666.selenidepom.pom.LangConditionedElement
import com.github.qky666.selenidepom.pom.Required
import com.github.qky666.selenidepom.pom.Widget
import com.github.qky666.selenidepom.pom.shouldLoadRequired

class CookiesBannerWidget(self: SelenideElement) : Widget(self) {

    val decline = LangConditionedElement(
        find("button#CybotCookiebotDialogBodyButtonDecline"),
        mapOf("es" to text("Denegar")),
        false,
    )
    @Required val accept = LangConditionedElement(
        find("button#CybotCookiebotDialogBodyLevelButtonLevelOptinAllowAll"),
        mapOf("es" to text("Permitir todas")),
        false,
    )

    fun acceptCookies() {
        shouldLoadRequired().accept.click()
        should(disappear)
    }
}
